Issue 1062 - Nudging

randomeizer opened the pull request:
Fixes #1062 

Currently added:
* You can get/set the ‘orientation’ of a ColorWell via `ColorWell:colorOrientation()`
* You can nudge the value using `ColorWell:nudge(right, up)`

For example:

```lua
local master = _fcp:inspector():color():colorWells():master()
master:nudgeColor(2*ColorWell.KEY_PRESS, 0)
```

This will nudge the color well 2 ‘key press’ values to the right. A key press being how far it would move if the well was selected and you used the shortcut to shift the color right, twice.

Not included:
* Hooking this up with Tangent

**ALSO NOTE:** This branch has merged the #970 branch to resolve some issues with `cp.prop` that were fixed in that branch. As such, I suggest merging #970 before this one.

randomeizer committed:
#1062 Rejigged the UI lookups

randomeizer committed:
#1062
* Added `ColorWell.colorOrientation` property to access the left/right/up/down orientation of the color wheel
* Removed several other `Position` properties to simplify

randomeizer committed:
Merge branch 'feature/issue-960' into feature/issue-1062

# Conflicts:
#	src/extensions/cp/prop/init.lua

randomeizer committed:
#1062 Added some default values for orientation modification

randomeizer committed:
#1062 Removed some debugging comments

randomeizer committed:
#1062 Fixed bug with `cp.prop:wrap(...)`

randomeizer committed:
#1062 Bug fixes.

randomeizer committed:
#1062
* First attempt at adding Master Color Wheel to Tangent. Not functioning for me yet...

latenitefilms commented:
FYI: This is the error I get when I try to use the Color Wheel controls with the Tangent:

```lua
2018-03-07 16:13:24: ********
2018-03-07 16:13:24: 16:13:24 ERROR:   LuaSkin: hs.socket read callback error: ...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:184: ERROR: incorrect type 'nil' for argument 1 (expected table)
stack traceback:
	[C]: in function 'hs.drawing.color.asHSB'
	...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:184: in function <...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:183>
	(...tail calls...)
	...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:945: in field '_get'
	...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:486: in function <...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:483>
	(...tail calls...)
	...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:490: in function 'cp.apple.finalcutpro.inspector.color.ColorWell.nudge'
	...ions/cp/apple/finalcutpro/inspector/color/ColorWheel.lua:306: in function 'cp.apple.finalcutpro.inspector.color.ColorWheel.nudgeColor'
	...andPost/src/plugins/finalcutpro/tangent/manager/init.lua:280: in field 'shiftValue'
	...andPost/src/plugins/finalcutpro/tangent/manager/init.lua:786: in function 'hs.tangent._callback'
	...st.app/Contents/Resources/extensions/hs/tangent/init.lua:339: in upvalue 'processHubCommand'
	...st.app/Contents/Resources/extensions/hs/tangent/init.lua:1475: in function <...st.app/Contents/Resources/extensions/hs/tangent/init.lua:1461>
2018-03-07 16:13:24: ********
```

randomeizer commented:
Hmm. That would indicate that it can't find the colour wheel. It shouldn't
be throwing an error because it the fact that it's nil should be getting
caught earlier. But my guess is the rooting the problem is finding the
colour well. If you do `master():UI()` does it come back with a result?
On Wed, 7 Mar 2018 at 3:16 pm, Chris Hocking <notifications@github.com>
wrote:

> FYI: This is the error I get when I try to use the Color Wheel controls
> with the Tangent:
>
> 2018-03-07 16:13:24: ********2018-03-07 16:13:24: 16:13:24 ERROR:   LuaSkin: hs.socket read callback error: ...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:184: ERROR: incorrect type 'nil' for argument 1 (expected table)
> stack traceback:
> 	[C]: in function 'hs.drawing.color.asHSB'
> 	...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:184: in function <...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:183>
> 	(...tail calls...)
> 	...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:945: in field '_get'
> 	...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:486: in function <...cking/Github/CommandPost/src/extensions/cp/prop/init.lua:483>
> 	(...tail calls...)
> 	...sions/cp/apple/finalcutpro/inspector/color/ColorWell.lua:490: in function 'cp.apple.finalcutpro.inspector.color.ColorWell.nudge'
> 	...ions/cp/apple/finalcutpro/inspector/color/ColorWheel.lua:306: in function 'cp.apple.finalcutpro.inspector.color.ColorWheel.nudgeColor'
> 	...andPost/src/plugins/finalcutpro/tangent/manager/init.lua:280: in field 'shiftValue'
> 	...andPost/src/plugins/finalcutpro/tangent/manager/init.lua:786: in function 'hs.tangent._callback'
> 	...st.app/Contents/Resources/extensions/hs/tangent/init.lua:339: in upvalue 'processHubCommand'
> 	...st.app/Contents/Resources/extensions/hs/tangent/init.lua:1475: in function <...st.app/Contents/Resources/extensions/hs/tangent/init.lua:1461>2018-03-07 16:13:24: ********
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CommandPost/CommandPost/pull/1064#issuecomment-371025610>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbEwzUry-N3cXMWYAtaq1ac9mnecWW6ks5tb20cgaJpZM4ScjE8>
> .
>

latenitefilms committed:
#1062

- Moved Tangent Manager into Core
- Moved Final Cut Pro specific code into Final Cut Pro Tangent Manager
- Updated Tangent UI Messages
- Updated Tangent Default Maps
- Closes #1062

latenitefilms committed:
Merge branch 'develop' into feature/issue-1062

latenitefilms committed:
#1062

- Removed Debugging
